### PR TITLE
update baidu cloudprovider

### PR DESCRIPTION
**Pull request by wccsama**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it:**
The performance of autoscaling group in baidu cloud provider is not as good as we
expected, so we update the cloud provider and fix some issues.

**Which issue(s) this PR fixes:**
None

**Does this PR introduce a user-facing change?:**
None

### CODE REVIEW

**Comment by hello2mao** [comment]
/assign @hello2mao
Thanks @wccsama

File: cluster-autoscaler/cloudprovider/baiducloud/baiducloud_auto_scaling_groups.go [hunk 1]
```diff
@@ -55,52 +55,46 @@ func (m *autoScalingGroups) Register(asg *Asg) {
...
+	m.instancesNotInManagedAsg[instanceID] = struct{}{}
 	return nil, nil
 }
 
 func (m *autoScalingGroups) regenerateCache() error {
-	newCache := make(map[BaiducloudRef]*Asg)
-
+	newCache := make(map[string]*Asg)
 	// TODO: Currently, baiducloud cloudprovider not support Multiple ASG
-	instanceList, err := m.cceClient.ListInstances(m.cloudConfig.ClusterID)
```
Comment (hello2mao) [inline cluster-autoscaler/cloudprovider/baiducloud/baiducloud_auto_scaling_groups.go]: Multiple ASG is supported?

File: cluster-autoscaler/cloudprovider/baiducloud/baiducloud_manager.go [hunk 2]
```diff
@@ -142,43 +144,29 @@ func randStringBytes(n int) string {
...
-	cceCluster, err := m.cceClient.DescribeCluster(m.cloudConfig.ClusterID)
-	if err != nil {
-		klog.Errorf("error while ScaleUpCluster since DescribeCluster failed.")
-		return fmt.Errorf("scaleUpCluster error: %v", err)
-	}
-	args = &cce.ScaleUpClusterArgs{
+// ScaleUpCluster Scale UP cluster
+func (m *BaiducloudManager) ScaleUpCluster(delta int, groupID string) error {
+	var args *cce.ScaleUpClusterWithGroupIDArgs
+	// password := randStringBytes(4) + "123!T"
```
Comment (hello2mao) [inline cluster-autoscaler/cloudprovider/baiducloud/baiducloud_manager.go]: Not used codes should be removed.

Comment (wccsama) [reply cluster-autoscaler/cloudprovider/baiducloud/baiducloud_manager.go]: ok

File: cluster-autoscaler/cloudprovider/baiducloud/baiducloud_auto_scaling_groups.go (hunk 1 above)
Comment (wccsama) [reply cluster-autoscaler/cloudprovider/baiducloud/baiducloud_auto_scaling_groups.go]: Multiple ASG is not supported currently, will be supported in the next version.

File: cluster-autoscaler/cloudprovider/baiducloud/baiducloud-sdk-go/cce/cluster.go [hunk 3]
```diff
@@ -348,14 +370,138 @@ func (c *Client) ScaleDownCluster(args *ScaleDownClusterArgs) error {
...
+	resp, err := c.SendRequest(req, nil)
+	if err != nil {
+		return err
+	}
+	bodyContent, err := resp.GetBodyContent()
+	if err != nil {
+		return err
+	}
+	var scResp *ScaleUpClusterResponse
+	err = json.Unmarshal(bodyContent, &scResp)
```
Comment (cwdsuzhou) [inline cluster-autoscaler/cloudprovider/baiducloud/baiducloud-sdk-go/cce/cluster.go]: take address of pointer? a bug?

Comment (wccsama) [reply cluster-autoscaler/cloudprovider/baiducloud/baiducloud-sdk-go/cce/cluster.go]: Yes, but does not affect the function. I will fix it Immediately.

**Pull request merged**